src/ucp/rma/rma_sw.c

```diff
@@ -15,9 +15,25 @@
 #include <ucs/profile/profile.h>
 #include <ucp/core/ucp_request.inl>
 
+#include <ucp/proto/proto_init.h>
 #include <ucp/proto/proto_common.inl>
 
 
+int ucp_rma_sw_proto_init_check_op(const ucp_proto_init_params_t *init_params,
+                                   uint64_t op_id_mask)
+{
+    ucp_context_h context = init_params->worker->context;
+
+    return (context->config.ext.rma_offload != UCS_YES) &&
+           ucp_proto_init_check_op(init_params, op_id_mask);
+}
+
+unsigned ucp_rma_sw_emulation_flag(ucp_context_h context)
+{
+    return (context->config.ext.rma_offload != UCS_NO) ?
+           UCP_PROTO_COMMON_INIT_FLAG_EMULATION : 0;
+}
```

Comment:
I think the protocol should exhibit the right flag regardless of ep config, and the common protocol selection logic should prefer offload according to disabled_proto_mask